packages/ampd-sdk/src/grpc/client.rs

```diff
@@ -143,6 +181,68 @@ impl From<Key> for KeyId {
     }
 }
 
+pub async fn with_timeout_and_retry<F, Fut, R>(
+    mut operation: F,
+    timeout_duration: Duration,
+    retry_policy: RetryPolicy,
```

Comment:
are we expecting to have different retry policies for different calls? maybe we can get rid of this parameter and use the constant directly